rust/src/common.rs

```diff
@@ -1,6 +1,23 @@
 use std::ffi::CString;
 use std::os::raw::c_char;
 
+pub mod nom7 {
+    use nom7::bytes::streaming::{tag, take_until};
+    use nom7::error::ParseError;
+    use nom7::IResult;
+
+    pub fn take_until_and_consume<'a, E: ParseError<&'a [u8]>>(t: &'a [u8])
+         -> impl Fn(&'a [u8]) -> IResult<&'a [u8], &'a [u8], E>
```

Comment:
I think this needs a comment. It wasn't clear to me that it consumes that terminator until I read the code.  My first that was "take_until" does consume...

Reply:
Ok, I'll add a comment to the function documentation